src/vs/workbench/contrib/terminal/browser/terminalProcessManager.ts

```diff
@@ -328,7 +333,20 @@ export class TerminalProcessManager extends Disposable implements ITerminalProce
 		const useConpty = this._configHelper.config.windowsEnableConpty && !isScreenReaderModeEnabled;
 		const shouldPersist = this._configHelper.config.enablePersistentSessions && !shellLaunchConfig.isFeatureTerminal;
 
-		this._terminalInstanceService.onPtyHostUnresponsive(() => this._onPtyDisconnect.fire());
+		this._register(this._terminalInstanceService.onPtyHostUnresponsive(() => {
+			this.isDisconnected = true;
+			this._onPtyDisconnect.fire();
+		}));
+		this._ptyResponsiveListener = this._terminalInstanceService.onPtyHostResponsive(() => {
```

Comment:
can't we do this.register of this so we don't need?https://github.com/microsoft/vscode/blob/f9e6b352d1cf054af74deb1bacb03e255ea4c52a/src/vs/workbench/contrib/terminal/browser/terminalProcessManager.ts#L344-L344

Reply:
It's set here because we need to dispose it and not react when the pty host gets restarted.